app/models/transaction_type.rb

```diff
@@ -0,0 +1,29 @@
+class TransactionType < ApplicationRecord
```

Comment:
Usually we would define
```ruby
enum operation: {
  credit: 'credit'.freeze,
  debit: 'debit'.freeze
}
```
in this file and it would prevent creating a `TransactionType` with an operation different than `debit` or `credit`.
Not sure if we can reuse the enum from `BankTransaction` like `enum operation: BankTransaction::operation`

Reply:
I didn't do that because the operations are effectively defined by `NAMES` - they are the keys! Defining them again in an `enum` would mean they were defined twice.

Reply:
I could do:
```ruby
enum operation: NAMES.keys.each_with_object({}) {|k,h| h[k] = k.to_s.freeze}
```